### PR TITLE
[CS Issue][Batch Plugin] Fix CS issue in Batch Plugin

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - the issue the static analyzer is complaining about is wait_for() not being in the loop
 - *...*

### Tickets:
 - CVS-145094

### CODE REVIEW

**Comment by zhaixuejun1993** [comment]
@ilya-lavrenov  can u help merge this PR for fixing CS issue?

File: src/plugins/auto_batch/src/compiled_model.cpp [hunk 1]
```diff
@@ -80,10 +80,8 @@ CompiledModel::GetWorkerInferRequest() const {
         workerRequestPtr->_thread = std::thread([workerRequestPtr, this] {
             while (1) {
                 std::cv_status status;
-                {
-                    std::unique_lock<std::mutex> lock(workerRequestPtr->_mutex);
-                    status = workerRequestPtr->_cond.wait_for(lock, std::chrono::milliseconds(m_time_out));
-                }
+                std::unique_lock<std::mutex> lock(workerRequestPtr->_mutex);
+                status = workerRequestPtr->_cond.wait_for(lock, std::chrono::milliseconds(m_time_out));
```
Comment (ilya-lavrenov) [inline src/plugins/auto_batch/src/compiled_model.cpp:84]: I suppose mutex should be released after wait_for, should not it?

Comment (songbell) [reply src/plugins/auto_batch/src/compiled_model.cpp:84]: brief look seems we don't need to move this out of the code block, what  is the original complain of this issue?

Comment (zhaixuejun1993) [reply src/plugins/auto_batch/src/compiled_model.cpp:84]: @songbell  @ilya-lavrenov 
the static analyzer is complaining about is wait_for() not being in the loop.
image

Comment (peterchen-intel) [reply src/plugins/auto_batch/src/compiled_model.cpp:84]: https://juniorfans.gitbooks.io/blog/content/3-condvar-spurious-wakeups.html

**Review by yangwang201911 — APPROVED** [review]

File: src/plugins/auto_batch/src/compiled_model.hpp [hunk 2]
```diff
@@ -30,6 +30,7 @@ class CompiledModel : public ov::ICompiledModel {
         std::condition_variable _cond;
         std::mutex _mutex;
         std::exception_ptr _exception_ptr;
+        bool _is_wakeup;
```
Comment (riverlijunjie) [inline src/plugins/auto_batch/src/compiled_model.hpp:33]:
```suggestion
       std::atomic<bool> _is_wakeup;
```
But current definition also is ok.

**Review by riverlijunjie — APPROVED** [review]
LGTM

File: src/plugins/auto_batch/src/compiled_model.cpp [hunk 3]
```diff
@@ -83,6 +85,8 @@ CompiledModel::GetWorkerInferRequest() const {
                 {
                     std::unique_lock<std::mutex> lock(workerRequestPtr->_mutex);
                     status = workerRequestPtr->_cond.wait_for(lock, std::chrono::milliseconds(m_time_out));
+                    if ((status != std::cv_status::timeout) && (workerRequestPtr->_is_wakeup == false))
```
Comment (songbell) [inline src/plugins/auto_batch/src/compiled_model.cpp:88]: I doubt this is useful in batch logic..._is_wakeup is set to true when the subrequested is fully collected and subrequests are done, but never reset to false, how do  you ensure no spurious wake up if second or third infer requests?

**Review by wangleis — APPROVED** [review]

**Pull request merged**